src/intervals/interval_functions.jl

```diff
@@ -1,95 +1,105 @@
-
 ## Powers
-# Integer power of an interval:
+# Integer power:
 function ^{T}(a::Interval{T}, n::Integer)
-    n < zero(n) && return inv( a^(-n) )
-    n == zero(n) && return one(a)
-    n == one(n) && return a
-    #
-    ## NOTE: square(x) is deprecated in favor of x*x
-    if n == 2*one(n)   # this is unnecessary as stands, but  mig(a)*mig(a) is supposed to be more efficient
-        return @round(T, mig(a)^2, mag(a)^2)
-    end
-    #
-    ## even power
-    if n%2 == 0
-        return @round(T, mig(a)^n, mag(a)^n)
-    end
-    ## odd power
+    n < 0  && return inv(a^(-n))
+    n == 0 && return one(a)
+    n == 1 && return a
+
+    iseven(n) && return @round(T, mig(a)^n, mag(a)^n)  # even power
+
+#     @show a.lo, a.hi
+#     @show a.lo^n, a.hi^n
+#     @show @round(T, a.lo^n, a.hi^n)
+
 
-    @round(T, a.lo^n, a.hi^n)
+    @round(T, a.lo^n, a.hi^n)  # odd power
 end
 
-^{T}(a::Interval{T}, r::Rational) = (a^(r.num)) ^ (1/r.den)
+function ^{T}(a::Interval{T}, r::Rational)
+
+    root = one(T)/r.den
+
+    if a.lo > zero(a.lo)
+        nth_root = @round(T, a.lo^root, a.hi^root)
+    else
+        if iseven(r.den)
+            nth_root = @round(T, mig(a)^root, mag(a)^root)
+        else
+            nth_root = @round(T, sign(a.lo)*abs(a.lo)^root, sign(a.hi)*abs(a.hi)^root)
+        end
+    end
+
+    nth_root^r.num
+end
 
 # Real power of an interval:
-function ^{T}(a::Interval{T}, x::Real)
-    isinteger(x) && return a^round(Int,x)
-    x < zero(x) && return inv( a^(-x) )
-    x == one(x)/2 && return sqrt(a)
-    #
-    z = zero(a.hi)
-    z > a.hi && error("Undefined operation; Interval is strictly negative and power is not an integer")
-    #
-    xInterv = convert(Interval{T}, x)  # @interval(x) ?
-    diam( xInterv ) >= eps(x) && return a^xInterv
+function ^{T}(a::Interval{T}, x::FloatingPoint)
+    isinteger(x)  && return a^(round(Int,x))
+    x < zero(x)  && return inv(a^(-x))
+    x == 0.5  && return sqrt(a)
```

Comment:
This relates to our long-lasting discussion on conversion... I think that instead of
`x == 0.5 ` it should be written `x == T(0.5)` or something alike, to avoid conversion.
I guess that this imposes some use of `@compat`, since it is a v0.4 usage.